kong/tools/batch_queue.lua

```diff
@@ -306,6 +311,10 @@ function Queue:flush()
   if current_batch_size > 0 then
     ngx_log(DEBUG, "queueing batch for processing (", current_batch_size, " entries)")
 
+    if #self.batch_queue == self.max_queued_batches then
+      ngx_log(ERR, "batch queue limit reached, dropping oldest batch")
+      remove(self.batch_queue, 1)
+    end
```

Comment:
Since we don't know exactly who is going to use batchqueue we have to think of it as a "pseudo library". ("lua-resty-batchqueue").

While the current interface doesn't allow creating more than a batch before triggering a flush, in theory at least the `max_queued_batches` parameter could be changed at run time. Better be a bit deffensive here:

```suggestion
    while #self.batch_queue <= self.max_queued_batches do
      ngx_log(ERR, "batch queue limit reached, dropping oldest batch")
      remove(self.batch_queue, 1)
    end
```